plugins/generator-1.21.4/neoforge-1.21.4/templates/block/blockentity.java.ftl

```diff
@@ -218,6 +243,110 @@ public class ${name}BlockEntity extends RandomizableContainerBlockEntity impleme
 	}
     </#if>
 
+    <#if data.sensitiveToVibration>
+    @Override public VibrationSystem.Data getVibrationData() {
+    	return this.vibrationData;
+    }
+
+    @Override public VibrationSystem.User getVibrationUser() {
+    	return this.vibrationUser;
+    }
+
+    @Override public VibrationSystem.Listener getListener() {
+    	return this.vibrationListener;
+    }
+
+	private class VibrationUser implements VibrationSystem.User {
+		private final int x;
+		private final int y;
+		private final int z;
+		private final BlockPos blockPos;
+
+		public VibrationUser(BlockPos blockPos) {
+			this.x = blockPos.getX();
+			this.y = blockPos.getY();
+			this.z = blockPos.getZ();
+			this.blockPos = blockPos;
+		}
+
+		@Override public PositionSource getPositionSource() {
+			return new BlockPositionSource(this.blockPos);
+		}
+
+		<#if data.vibrationalEvents?has_content>
+		@Override public TagKey<GameEvent> getListenableEvents() {
+			return TagKey.create(Registries.GAME_EVENT, ResourceLocation.withDefaultNamespace("${data.getModElement().getRegistryName()}_can_listen"));
+		}
+		</#if>
+
+		@Override public int getListenerRadius() {
+			<#if hasProcedure(data.vibrationSensitivityRadius)>
+				Level world = ${name}BlockEntity.this.getLevel();
+				BlockState blockstate = world.getBlockState(this.blockPos);
+				return (int) <@procedureOBJToNumberCode data.vibrationSensitivityRadius/>;
+			<#else>
+				return ${data.vibrationSensitivityRadius.getFixedValue()};
+			</#if>
+		}
+
+		@Override public boolean isValidVibration(Holder<GameEvent> holder, GameEvent.Context context) {

```

Comment:
As noticed by @Spectrall368, this will only make sneaking ignored for certain vibration types due to tag check. Does it even make sense to allow to override this system by the user? I guess if the vibration is set to be ignored when sneaking, one would probably expect that it is in this case ignored by all receivers